mmv1/products/vertexai/Models.yaml

```diff
@@ -0,0 +1,425 @@
+              * L1_DISTANCE: Manhattan (L_1) Distance
+              * COSINE_DISTANCE: Cosine Distance. Defined as 1 - cosine similarity.
+              * DOT_PRODUCT_DISTANCE: Dot Product Distance. Defined as a negative of the dot product
+            default_value: 'DOT_PRODUCT_DISTANCE'
+          - type: String
+            name: 'featureNormType'
+            description: |-
+              Type of normalization to be carried out on each vector. The value must be one of the followings:
+              * UNIT_L2_NORM: Unit L2 normalization type
+              * NONE: No normalization type is specified.
+            default_value: 'NONE'
+            ignore_read: true
+          - type: NestedObject
+            name: 'algorithmConfig'
+            description:
+              The configuration with regard to the algorithms used for efficient
+              search.
+            properties:
+              - type: NestedObject
+                name: 'treeAhConfig'
+                exactly_one_of:
+                  - treeAhConfig
+                  - bruteForceConfig
+                description: |-
+                  Configuration options for using the tree-AH algorithm (Shallow tree + Asymmetric Hashing).
+                  Please refer to this paper for more details: https://arxiv.org/abs/1908.10396
+                properties:
+                  - type: Integer
+                    name: 'leafNodeEmbeddingCount'
+                    description:
+                      Number of embeddings on each leaf node. The default value
+                      is 1000 if not set.
+                    default_value: 1000
+                  - type: Integer
+                    name: 'leafNodesToSearchPercent'
+                    description: |-
+                      The default percentage of leaf nodes that any query may be searched. Must be in
+                      range 1-100, inclusive. The default value is 10 (means 10%) if not set.
+                    default_value: 10
+              - type: NestedObject
+                name: 'bruteForceConfig'
+                allow_empty_object: true
+                send_empty_value: true
+                properties: []
+                exactly_one_of:
+                  - treeAhConfig
+                  - bruteForceConfig
+                description: |-
+                  Configuration options for using brute force search, which simply implements the
+                  standard linear search in the database for each query.
+  - type: String
+    name: 'sourceModel'
+    default_from_api: true
+    at_least_one_of:
+      - model
+      - sourceModel
+    description: |
+      The resource name of the Model to copy. That Model must be in the same Project
+  - type: ResourceRef
+    name: 'model'
+    resource: 'Model'
+    imports: 'displayName'
+    at_least_one_of:
+      - sourceModel
+      - model
+    description: |
+      The Model to create.
```

Comment:
I believe this field is only used for create-via-upload. Could you please remove this field (and any other fields specific to create-via-upload) from this PR and include them in your next PR that adds the new creation method?

Reply:
I removed as much as I could but source_model does depend on several fields since it needs a place to store the incoming fields that it it's copying from another model.

The response when a copy method is sent:
```hcl
{
    "name": "projects/871647908372/locations/us-central1/models/4741205189644517376",
    "displayName": "sampled_shapley_upload_container_spec",
    "predictSchemata": {},
    "containerSpec": {
        "imageUri": "us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-12:latest",
        "env": [
            {
                "name": "TF_CPP_MIN_LOG_LEVEL",
                "value": "2"
            },
            {
                "name": "AIP_STORAGE_URI",
                "value": "gs://your-bucket/model"
            }
        ],
        "ports": [
            {
                "containerPort": 8080
            }
        ],
        "predictRoute": "/v1/models/model/predict",
        "healthRoute": "/health",
        "grpcPorts": [
            {
                "containerPort": 8500
            }
        ],
        "deploymentTimeout": "300s",
        "sharedMemorySizeMb": "1024",
        "startupProbe": {
            "exec": {
                "command": [
                    "curl",
                    "http://localhost:8080/health"
                ]
            },
            "periodSeconds": 30,
            "timeoutSeconds": 10
        },
        "healthProbe": {
            "exec": {
                "command": [
                    "curl",
                    "http://localhost:8080/health"
                ]
            },
            "periodSeconds": 30,
            "timeoutSeconds": 5
        }
    },
    "supportedDeploymentResourcesTypes": [
        "DEDICATED_RESOURCES",
        "SHARED_RESOURCES"
    ],
    "supportedInputStorageFormats": [
        "jsonl",
        "bigquery",
        "csv",
        "tf-record",
        "tf-record-gzip",
        "file-list"
    ],
    "supportedOutputStorageFormats": [
        "jsonl",
        "bigquery"
    ],
    "createTime": "2024-10-31T01:08:20.493644Z",
    "updateTime": "2024-10-31T01:08:27.098115Z",
    "etag": "AMEw9yPNOVVwS7j2Pyg7Fo-1eJ6fAuJz24zjmF5JqdeXBAPXM8Ysr8nZtbIXW5bjoIc=",
    "supportedExportFormats": [
        {
            "id": "custom-trained",
            "exportableContents": [
                "ARTIFACT"
            ]
        }
    ],
    "explanationSpec": {
        "parameters": {
            "sampledShapleyAttribution": {
                "pathCount": 25
            }
        },
        "metadata": {}
    },
    "artifactUri": "gs://cloud-samples-data/ai-platform/mnist_tfrecord/pretrained",
    "versionId": "1",
    "versionAliases": [
        "default"
    ],
    "versionCreateTime": "2024-10-31T01:08:20.493644Z",
    "versionUpdateTime": "2024-10-31T01:08:27.053530Z",
    "originalModelInfo": {
        "model": "projects/871647908372/locations/us-central1/models/7131490691871408128@1"
    },
    "modelSourceInfo": {
        "sourceType": "CUSTOM",
        "copy": true
    }
}
```